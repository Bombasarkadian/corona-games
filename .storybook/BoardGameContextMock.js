import React from "react";
import { boolean, number, text, select } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { BoardGameProvider } from "contexts/BoardGameContext";

const ref = "Game Context";

const BoardGameContextMock = ({ children }) => {
  return (
    <BoardGameProvider
      G={{
        timePerTurn: 120,
        turnEndTime:
          Math.floor(new Date().getTime() / 1000) +
          number(
            "Remaining seconds",
            100,
            {
              range: true,
              min: 0,
              max: 120,
              step: 1,
            },
            ref
          ),
        points: [10, 3],
        actions: [
          { action: "message", text: text("Message text", "Hello there, shall we begin?", ref) },
        ],
        privateRoom: boolean("Private", true, ref),
        players: {
          "0": {
            phrase: text("Phrase", "Baba z wozu", ref),
            card: { pictures: [], layout: 0, rotation: 0 },
          },
          "1": {},
        },
        canChangePhrase: boolean("Can change phrase?", true, ref),
        currentCard: { pictures: [], layout: 0, rotation: 0 },
      }}
      ctx={{
        activePlayers: [
          select("stage", ["draw", "guess", "wait", "manage", "match"], "draw", ref),
          "guess",
        ],
        phase: select("phase", ["wait", "play"], "play", ref),
        numPlayers: number("Player count", 4, { range: true, min: 2, max: 10, step: 1 }, ref),
      }}
      moves={{
        Guess: action("Guess"),
        Ping: action("Ping"),
        UpdateDrawing: action("UpdateDrawing"),
        NotifyTimeout: action("NotifyTimeout"),
      }}
      playerID={0}
      gameName={"Kalambury"}
      gameID={"qwe123"}
      rawClient={{ transport: { socket: null } }}
      gameMetadata={[
        { id: 0, name: "user-0", isConnected: true },
        { id: 1, name: "user-1", isConnected: false },
      ]}
    >
      {children}
    </BoardGameProvider>
  );
};

export default BoardGameContextMock;
