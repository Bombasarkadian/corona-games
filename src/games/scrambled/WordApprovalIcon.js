import React from "react";
import PropTypes from "prop-types";
import { Icon } from "semantic-ui-react";
import { availableLaguages } from "./data/tiles";
import { useBoardGame } from "contexts/BoardGameContext";

const propTypes = {
  letters: PropTypes.arrayOf(PropTypes.string).isRequired,
  checkedIn: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const WordApprovalIcon = ({ letters, checkedIn }) => {
  const {
    G: { language, assist },
  } = useBoardGame();
  const currentLanguage = availableLaguages.find(({ key }) => key === language);

  if (!checkedIn.includes(assist) || letters.length > 3)
    return <Icon name="circle dot outline" color="blue" />;

  if (currentLanguage.shorts.includes(letters.join(""))) {
    return <Icon name="check circle" color="green" />;
  } else {
    return <Icon name="remove circle" color="red" />;
  }
};

WordApprovalIcon.propTypes = propTypes;
export default WordApprovalIcon;
