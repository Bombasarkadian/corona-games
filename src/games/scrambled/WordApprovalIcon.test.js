import React from "react";
import renderer from "react-test-renderer";
import BoardGameContextMock from "./GameContextMock";
import WordApprovalIcon from "./WordApprovalIcon";

describe("scrambled WordApprovalIcon component", () => {
  it("renders correctly for valid word", () => {
    expect.hasAssertions();
    let component;
    renderer.act(() => {
      component = renderer.create(
        <BoardGameContextMock>
          <WordApprovalIcon letters={["H", "E", "Y"]} checkedIn={["approval"]} />
        </BoardGameContextMock>
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly for invalid word", () => {
    expect.hasAssertions();
    let component;
    renderer.act(() => {
      component = renderer.create(
        <BoardGameContextMock>
          <WordApprovalIcon letters={["Z", "X", "W"]} checkedIn={["approval"]} />
        </BoardGameContextMock>
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly for long word", () => {
    expect.hasAssertions();
    let component;
    renderer.act(() => {
      component = renderer.create(
        <BoardGameContextMock>
          <WordApprovalIcon letters={["Z", "X", "W", "A", "B"]} checkedIn={["approval"]} />
        </BoardGameContextMock>
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with disabled checks", () => {
    expect.hasAssertions();
    let component;
    renderer.act(() => {
      component = renderer.create(
        <BoardGameContextMock>
          <WordApprovalIcon letters={["Z", "X", "W"]} checkedIn={[]} />
        </BoardGameContextMock>
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
