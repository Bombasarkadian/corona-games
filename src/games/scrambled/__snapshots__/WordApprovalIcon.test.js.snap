// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrambled WordApprovalIcon component renders correctly for invalid word 1`] = `
<i
  aria-hidden="true"
  className="red remove circle icon"
  onClick={[Function]}
/>
`;

exports[`scrambled WordApprovalIcon component renders correctly for long word 1`] = `
<i
  aria-hidden="true"
  className="blue circle dot outline icon"
  onClick={[Function]}
/>
`;

exports[`scrambled WordApprovalIcon component renders correctly for valid word 1`] = `
<i
  aria-hidden="true"
  className="green check circle icon"
  onClick={[Function]}
/>
`;

exports[`scrambled WordApprovalIcon component renders correctly with disabled checks 1`] = `
<i
  aria-hidden="true"
  className="blue circle dot outline icon"
  onClick={[Function]}
/>
`;
